spec/friendly_shipping/services/usps_ship_spec.rb

```diff
@@ -162,4 +162,44 @@
       end
     end
   end
+
+  describe "#timings" do
+    subject(:timings) { service.timings(shipment, options: options) }
+
+    let(:destination) { FactoryBot.build(:physical_location, region: "FL", zip: '32821') }
+    let(:origin) { FactoryBot.build(:physical_location, region: "NC", zip: '27703') }
+    let(:shipment) { FactoryBot.build(:physical_shipment, packages: [], origin: origin, destination: destination) }
+
+    context "with a valid request", vcr: { cassette_name: "usps_ship/timings/success" } do
+      it { is_expected.to be_success }
+
+      it "has all the right data" do
+        result = timings.value!.data
+        expect(result.length).to eq(2)
+        timing = result.first
+        expect(timing).to be_a(FriendlyShipping::Timing)
+        expect(timing.pickup).to eq(Time.parse("2024-04-01 08:00"))
+        expect(timing.delivery).to eq(Time.parse("2024-04-04 18:00"))
+        expect(timing.guaranteed).to be(false)
+        expect(timing.data).to eq(
+          notes: "WEIGHT_LESS_THAN_1_POUND",
+          service_standard: "3",
+          service_standard_message: "3 Days"
+        )
+      end
+    end
+
+    context "when the access token is wrong", vcr: { cassette_name: "usps_ship/timings/failure" } do
+      let(:access_token) do
+        FriendlyShipping::Services::USPSShip::AccessToken.new(
+          token_type: "Bearer",
+          expires_in: 3599,
+          raw_token: "WRONG_TOKEN"
+        )
+      end
+
+      it { is_expected.to be_failure }
+      it { expect(timings.failure.to_s).to eq("Missing or malformed access token.") }
+    end
+  end
```

Comment:
What does USPS return for unknown zip codes?

Reply:
For timings, I get a response with a blank arrival date. For rates, I get this:

```
{
    "apiVersion": "/prices/v3",
    "error": {
        "code": "400",
        "message": "Could not find zone from provided ZIPs, date, and mail class",
        "errors": []
    }
}
```

Probably a good idea to add specs for both scenarios. I'll do that.

Reply:
I've added specs for both scenarios.

The timings API returns a successful response even with an invalid destination zip code so I'm handling that scenario in the response parser by detecting the missing delivery estimate and returning a failure.

The rates API returns a failed response (400) with an invalid destination zip code so I don't have to handle that in the parser. We can rely on the service class returning a failure.